Framework/Core/include/Framework/TableBuilder.h

```diff
@@ -833,7 +835,7 @@ auto spawner(framework::pack<C...> columns, arrow::Table* atable, const char* na
   for (auto i = 0u; i < sizeof...(C); ++i) {
     arrays.push_back(std::make_shared<arrow::ChunkedArray>(chunks[i]));
   }
-
+  new_schema = new_schema->WithMetadata(std::make_shared<arrow::KeyValueMetadata>(std::vector{std::string{"label"}}, std::vector{std::string{name}}));
```

Comment:
You can probably have a non templated helper function for this, which allows you not exposing the dependency on the key value stuff.

Reply:
True, this can go in TableBuilder.cxx